lib/activerecord_spanner_adapter/information_schema.rb

```diff
@@ -240,6 +240,27 @@ def foreign_keys table_name
       end
     end
 
+    def check_constraints table_name
+      sql = <<~SQL.squish
+        SELECT tc.TABLE_NAME,
+               tc.CONSTRAINT_NAME,
+               cc.CHECK_CLAUSE
+        FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS tc
+        INNER JOIN INFORMATION_SCHEMA.CHECK_CONSTRAINTS cc ON tc.CONSTRAINT_NAME = cc.CONSTRAINT_NAME
+        WHERE tc.TABLE_NAME = %<table_name>s
+          AND tc.CONSTRAINT_TYPE = 'CHECK'
+          AND tc.CONSTRAINT_NAME LIKE 'chk_rails_%%'
```

Comment:
It is a little funky because Rails uses `chk_rails` to [ignore dumping to schema](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/schema_dumper.rb#L27-L29) as that is the generated constraint name when a name is not given. 

I initially went the direction of ignoring the not null constraint but was a bit nervous if there were other system constraints that I had not encountered yet with the schema I am working with. If that is not the case, then I can update to just ignore the not null constraint otherwise we could introduce a setting like AR does to match an `chk_include_pattern` that is configurable for the application (or update `excl_ignore_pattern` with the not null as the default and backport to Rails 6.1). Thoughts?

Reply:
To my knowledge, `NOT NULL` constraints are the only system-managed check constraints that will show up here, so I think we can safely go with just excluding those. I think that is better as it will then work automatically for everyone, regardless of the naming conventions they might be using.